src/CachePeer.cc

```diff
@@ -14,9 +14,17 @@
 #include "pconn.h"
 #include "PeerPoolMgr.h"
 #include "SquidConfig.h"
+#include "util.h"
 
 CBDATA_CLASS_INIT(CachePeer);
 
+CachePeer::CachePeer(const char * const hostname):
+    name(xstrdup(hostname)),
+    host(xstrdup(hostname))
```

Comment:
FWIW, I started by changing `name` type to be SBuf instead of raw c-string, but had to revert all those improvements because they triggered too many source code changes, most of which were out of scope and some could not even be completed in this PR. That conversion and more CachePeer polishing changes should be done, but we should not do it now/here.